server/datastore/mysql/policies_test.go

```diff
@@ -2429,3 +2446,61 @@ func testCountPolicies(t *testing.T, ds *Datastore) {
 	require.NoError(t, err)
 	assert.Equal(t, 10, globalCount)
 }
+
+func testUpdatePolicyHostCounts(t *testing.T, ds *Datastore) {
+	// new policy
+	policy, err := ds.NewGlobalPolicy(context.Background(), nil, fleet.PolicyPayload{Name: "global policy 1"})
+	require.NoError(t, err)
+
+	team, err := ds.NewTeam(context.Background(), &fleet.Team{Name: "team1"})
+	require.NoError(t, err)
+
+	// create 4 team hosts
+	var teamHosts []*fleet.Host
+	for i := 0; i < 4; i++ {
+		h, err := ds.NewHost(context.Background(), &fleet.Host{OsqueryHostID: ptr.String(fmt.Sprintf("host%d", i)), NodeKey: ptr.String(fmt.Sprintf("host%d", i)), TeamID: &team.ID})
+		require.NoError(t, err)
+		teamHosts = append(teamHosts, h)
+	}
+
+	// create 4 global hosts
+	var globalHosts []*fleet.Host
+	for i := 4; i < 8; i++ {
+		h, err := ds.NewHost(context.Background(), &fleet.Host{OsqueryHostID: ptr.String(fmt.Sprintf("host%d", i)), NodeKey: ptr.String(fmt.Sprintf("host%d", i)), TeamID: nil})
+		require.NoError(t, err)
+		globalHosts = append(globalHosts, h)
+	}
+
+	// add policy responses
+	for _, h := range teamHosts {
+		res := map[uint]*bool{
+			policy.ID: ptr.Bool(true),
+		}
+		err = ds.RecordPolicyQueryExecutions(context.Background(), h, res, time.Now(), false)
+		require.NoError(t, err)
+	}
+
+	for _, h := range globalHosts {
+		res := map[uint]*bool{
+			policy.ID: ptr.Bool(true),
+		}
+		err = ds.RecordPolicyQueryExecutions(context.Background(), h, res, time.Now(), false)
+		require.NoError(t, err)
+	}
+
+	// check policy host counts before update
+	policy, err = ds.Policy(context.Background(), policy.ID)
+	require.NoError(t, err)
+	require.Equal(t, uint(0), policy.FailingHostCount)
+	require.Equal(t, uint(0), policy.PassingHostCount)
+
+	// update policy host counts
+	err = ds.UpdateHostPolicyCounts(context.Background())
+	require.NoError(t, err)
+
+	// check policy host counts
+	policy, err = ds.Policy(context.Background(), policy.ID)
+	require.NoError(t, err)
+	require.Equal(t, uint(0), policy.FailingHostCount)
+	require.Equal(t, uint(8), policy.PassingHostCount)
+}
```

Comment:
@mostlikelee Can you add a check for `TeamPolicy` here? The global policy is inhered by the team, right? So when retrieving policy results for the team we should get 4 passing hosts, right?

```
	// check team policy host counts
	policy, err = ds.TeamPolicy(context.Background(), team.ID, policy.ID)
	require.NoError(t, err)
	require.Equal(t, uint(0), policy.FailingHostCount)
	require.Equal(t, uint(4), policy.PassingHostCount)
```

But the above code doesn't work for me. Maybe I'm missing something.

Reply:
we only use inherited counts when listing policies for teams in the UI (https://dogfood.fleetdm.com/policies/manage?order_key=name&order_direction=asc&page=0&team_id=9)

`ds.TeamPolicy()` shouldn't return any inherited counts, they should only be returned in `ds.ListTeamPolicies`

Reply:
I'm not aware of fleet returning inherited counts for single policy lookups in the API currently

Reply:
Ok, looks like I got the method wrong. I still recommend adding a check for inherited policy count here, unless you think it is covered well elsewhere.

```
	// check inherited policies
	ctx := context.Background()
	inheritedPolicies, err := getInheritedPoliciesForTeam(ctx, ds.reader(ctx), team.ID, fleet.ListOptions{})
	require.NoError(t, err)
	require.Equal(t, 1, len(inheritedPolicies))
	require.Equal(t, uint(0), inheritedPolicies[0].FailingHostCount)
	require.Equal(t, uint(4), inheritedPolicies[0].PassingHostCount)
```

Reply:
I did have to update some existing tests using `ds.ListPolicies()` but worth a review

Reply:
it looks like coverage is pretty thorough here: https://github.com/fleetdm/fleet/blob/b2899c402eea31c8a9fd98375eaca3f84894d1d9/server/datastore/mysql/policies_test.go#L206

The only addition my test covers is the 0 case when the cron hasn't yet run